semcore/d3-chart/src/a11y/PlotA11yModule.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import { DataStructureHints, PartialDataSummarizationConfig } from './hints';
+import { normalizeLocale } from './locale';
+import { translations } from './translations/module/translations';
+import { Root, sstyled } from '@semcore/core';
+import styles from '../style/plotA11yModule.shadow.css';
+
+let globalWasFocused = false;
+let globalNavWithKeyboard = false;
+
+export type A11yViewProps = {
+  id: string;
+  data: Record<string, unknown>[];
+  hints: DataStructureHints;
+  plotLabel: string;
+  locale: NavigatorLanguage['language'];
+  config: PartialDataSummarizationConfig;
+  // eslint-disable-next-line ssr-friendly/no-dom-globals-in-module-scope
+  plotRef: React.RefObject<Element>;
+};
+
+export const PlotA11yModule: React.FC<A11yViewProps> = (props) => {
+  const SPlotA11yModule = Root;
+  const [wasFocused, setWasFocused] = React.useState(globalWasFocused);
+  const [navWithKeyboard, setNavWithKeyboard] = React.useState(globalNavWithKeyboard);
+  const [plotA11yView, setPlotA11yView] = React.useState<{
+    Component: React.FC<A11yViewProps>;
+  } | null>(null);
+
+  const hadnleHiddenElementsFocus = React.useCallback(() => {
+    setWasFocused(true);
+    setNavWithKeyboard(true);
+  }, []);
+
+  const [loading, setLoading] = React.useState(false);
+  const [error, setError] = React.useState<Error | null>(null);
+
+  const locale = React.useMemo(() => normalizeLocale(props.locale, translations), [props.locale]);
```

Comment:
Let's also take language out of context. Many applications are already wrapped in this.
import { I18nConsumer } from '@semcore/utils/lib/enhances/I18nConsumer';